contracts/vortex/CarbonVortex.sol

```diff
@@ -105,31 +172,171 @@ contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable,
+     */
+    function setPriceResetMultiplier(
+        uint32 newPriceResetMultiplier
+    ) external onlyAdmin greaterThanZero(newPriceResetMultiplier) {
+        _setPriceResetMultiplier(newPriceResetMultiplier);
+    }
+
+    /**
+     * @notice sets the minimum token sale amount multiplier
+     *
+     * Requirements:
+     *
+     * - The caller must be the admin of the contract.
+     */
+    function setMinTokenSaleAmountMultiplier(
+        uint32 newMinTokenSaleAmountMultiplier
+    ) external onlyAdmin greaterThanZero(newMinTokenSaleAmountMultiplier) {
+        _setMinTokenSaleAmountMultiplier(newMinTokenSaleAmountMultiplier);
+    }
+
+    /**
+     * @notice sets the price decay half-life for all tokens except target
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setPriceDecayHalfLife(
+        uint32 newPriceDecayHalfLife
+    ) external onlyAdmin greaterThanZero(newPriceDecayHalfLife) {
+        _setPriceDecayHalfLife(newPriceDecayHalfLife);
+    }
+
+    /**
+     * @notice sets the price decay half-life for the target token
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setTargetTokenPriceDecayHalfLife(
+        uint32 newPriceDecayHalfLife
+    ) external onlyAdmin greaterThanZero(newPriceDecayHalfLife) {
+        _setTargetTokenPriceDecayHalfLife(newPriceDecayHalfLife);
+    }
+
+    /**
+     * @notice sets the price decay half-life for the target token on reset
+     *
+     * Requirements:
+     *
+     * - The caller must be the admin of the contract.
+     */
+    function setTargetTokenPriceDecayHalfLifeOnReset(
+        uint32 newPriceDecayHalfLife
+    ) external onlyAdmin greaterThanZero(newPriceDecayHalfLife) {
+        _setTargetTokenPriceDecayHalfLifeOnReset(newPriceDecayHalfLife);
+    }
+
+    /**
+     * @notice sets the max (or initial) target token sale amount
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setMaxTargetTokenSaleAmount(
+        uint128 newMaxTargetTokenSaleAmount
+    ) external onlyAdmin greaterThanZero(newMaxTargetTokenSaleAmount) {
+        _setMaxTargetTokenSaleAmount(newMaxTargetTokenSaleAmount);
+    }
+
+    /**
+     * @notice sets the min target token sale amount
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setMinTargetTokenSaleAmount(
+        uint128 newMinTargetTokenSaleAmount
+    ) external onlyAdmin greaterThanZero(newMinTargetTokenSaleAmount) {
+        _setMinTokenSaleAmount(_targetToken, newMinTargetTokenSaleAmount);
+    }
+
+    /**
+     * @notice sets if trading is enabled or disabled for a token
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function disablePair(Token token) external onlyAdmin {
```

Comment:
This should be explicit (instead of triggering). Triggering is a bad UX due to human error (e.g., the DAO sends multiple transactions accidentally and not realizing that the state won't end up as intended)

Reply:
fixed